src/api/modelTypes.js

```diff
@@ -655,63 +674,76 @@ export type Submessage = $ReadOnly<{|
  * `Message`: `PmMessage` and `StreamMessage`.
  */
 type MessageBase = $ReadOnly<{|
-  /** Our own flag; if true, really type `Outbox`. */
-  isOutbox: false,
-
-  /**
-   * These don't appear in `message` events, but they appear in a `/message`
-   * response when a search is involved.
-   */
-  match_content?: string,
-  match_subject?: string,
-
-  /**
-   * The `flags` story is a bit complicated:
-   *  * Absent in `event.message` for a `message` event... but instead
-   *    (confusingly) there is an `event.flags`.
-   *  * Present under `EVENT_NEW_MESSAGE`.
-   *  * Present under `MESSAGE_FETCH_COMPLETE`, and in the server `/message`
-   *    response that action is based on.
-   *  * Absent in the Redux `state.messages`; we move the information to a
-   *    separate subtree `state.flags`.
-   */
-  flags?: $ReadOnlyArray<string>,
-
-  /**
-   * Present under EVENT_NEW_MESSAGE and under MESSAGE_FETCH_COMPLETE,
-   * and in `state.messages`, all as an AvatarURL, because we
-   * translate into that form at the edge.
-   *
-   * For how it appears at the edge (and how we translate) see
-   * AvatarURL.fromUserOrBotData.
-   */
-  avatar_url: AvatarURL,
-
-  // The rest are believed to really appear in `message` events.
+  //
+  // Properties on `event.message` in `message` events. Order follows the
+  // doc. See comment on Message for how up-to-date these are.
+  //
 
   client: string,
   content: string,
   content_type: 'text/html',
   edit_history: $ReadOnlyArray<MessageEdit>,
```

Comment:
While reordering to match the doc, I think it'd be helpful to also insert mentions here of the properties that do appear in the doc and aren't included in this type:
```suggestion
  content_type: 'text/html',
  // display_recipient handled on PmMessage and StreamMessage separately
  edit_history: $ReadOnlyArray<MessageEdit>,
```

Similarly later
```
  // stream_id handled on StreamMessage
```
and some others.

That just further helps make it easy to compare the type definition to the doc.